### PR TITLE
added Akka.Analyzer reference

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Changes

Adding https://github.com/akkadotnet/akka.analyzers support into Akka.NET itself.

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Design discussion issue #https://github.com/akkadotnet/akka.net/issues/3377
* [ ] I have added website documentation for this feature.

### CODE REVIEW

File: src/Akka.sln [hunk 1]
```diff
@@ -56,6 +56,8 @@ EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Build", "Build", "{10C5B1E8-15B5-4EB3-81AE-1FC054FE1305}"
 	ProjectSection(SolutionItems) = preProject
 		..\build.fsx = ..\build.fsx
+		Directory.Build.props = Directory.Build.props
```
Comment (Aaronontheweb) [inline src/Akka.sln:59]: Added these to solution file so they would be easier to edit.

File: src/Directory.Build.props [hunk 2]
```diff
@@ -69,6 +69,9 @@ If you want to see the [full set of changes made in Akka.NET v1.5.14, click here
 | 1       | 227  | 25   | Drew                |
 | 1       | 1    | 1    | szaliszali          |</PackageReleaseNotes>
   </PropertyGroup>
+  <ItemGroup Label="Analyzers">
+    <PackageReference Include="Akka.Analyzers" Version="0.1.2" PrivateAssets="all" />
```
Comment (Aaronontheweb) [inline src/Directory.Build.props:73]: Included this package as a private asset in `Directory.Build.props` to the analysis would be applied solution-wide.

**Comment by Aaronontheweb** [comment]
Was able to roll the changes back to the MSFT.EXT version range - turns out I had to upgrade to .NET 7.0 originally only because I have screwed up the NuGet package format for the Roslyn Analyzers themselves. Fixed that in https://github.com/akkadotnet/akka.analyzers/releases/tag/0.1.1 and was able to roll it back.

File: src/core/Akka.Cluster/Routing/ClusterRoutingConfig.cs [hunk 3]
```diff
@@ -298,7 +298,9 @@ public override int GetNrOfInstances(ActorSystem system)
         /// <inheritdoc cref="RouterConfig.CreateRouterActor"/>
         public override ActorBase CreateRouterActor()
         {
+#pragma warning disable AK1000
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster/Routing/ClusterRoutingConfig.cs:301]: Disabling `AK1000` here because custom router types kind of fall outside the normal use cases for not `new`-ing up actors

File: src/core/Akka.Cluster/Routing/ClusterRoutingConfig.cs [hunk 4]
```diff
@@ -298,7 +298,9 @@ public override int GetNrOfInstances(ActorSystem system)
...
             return new ClusterRouterPoolActor(Local.SupervisorStrategy, Settings);
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster/Routing/ClusterRoutingConfig.cs:302]: Disabling `AK1000` here because custom router types kind of fall outside the normal use cases for not `new`-ing up actors

File: src/core/Akka.Docs.Tests/Streams/StreamRefsDocTests.cs
Comment (Aaronontheweb) [inline src/core/Akka.Docs.Tests/Streams/StreamRefsDocTests.cs:1]: Applying `AK1001` fixes here (`PipeTo` closures)

File: src/core/Akka.Docs.Tests/Utilities/CircuitBreakerDocSpec.cs
Comment (Aaronontheweb) [inline src/core/Akka.Docs.Tests/Utilities/CircuitBreakerDocSpec.cs:1]: Applying `AK1001` fixes here (`PipeTo` closures)

File: src/core/Akka.Streams.Tests/Dsl/StreamRefsSerializerSpec.cs
Comment (Aaronontheweb) [inline src/core/Akka.Streams.Tests/Dsl/StreamRefsSerializerSpec.cs:1]: Applying `AK1001` fixes here (`PipeTo` closures)

File: src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs
Comment (Aaronontheweb) [inline src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs:1]: Applying `AK1001` fixes here (`PipeTo` closures)

File: src/core/Akka.Tests/Actor/ActorRefSpec.cs [hunk 5]
```diff
@@ -179,7 +179,7 @@ public async Task An_ActorRef_should_support_nested_ActorOfs()
             var a = Sys.ActorOf(Props.Create(() => new NestingActor(Sys)));
             a.Should().NotBeNull();
             
-            Func<Task> t1 = async () =>
+            var t1 = async () =>
```
Comment (Aaronontheweb) [inline src/core/Akka.Tests/Actor/ActorRefSpec.cs:182]: Unrelated fix

File: src/core/Akka.TestKit/Akka.TestKit.csproj [hunk 6]
```diff
@@ -29,7 +29,7 @@
         <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="$(MicrosoftLibVersion)" />
         <PackageReference Include="Nito.AsyncEx.Coordination" Version="5.1.2" />
         <PackageReference Include="Nito.AsyncEx.Context" Version="5.1.2" />
-        <PackageReference Include="System.Linq.Async" Version="$(MicrosoftLibVersion)" />
+        <PackageReference Include="System.Linq.Async" Version="6.0.1" />
```
Comment (Aaronontheweb) [inline src/core/Akka.TestKit/Akka.TestKit.csproj:32]: System.Linq.Async is actually part of Reactive Extensions (Rx) so it doesn't ship at the normal / same cadences as the rest of the MSFT libraries, so we shouldn't include it in the version range.

File: src/core/Akka.Tests/Dispatch/AsyncAwaitSpec.cs
Comment (Aaronontheweb) [inline src/core/Akka.Tests/Dispatch/AsyncAwaitSpec.cs:1]: Applying `AK1001` fixes here (`PipeTo` closures)

File: src/core/Akka.Tests/Routing/RoutingSpec.cs
Comment (Aaronontheweb) [inline src/core/Akka.Tests/Routing/RoutingSpec.cs:1]: Applying `AK1001` fixes here (`PipeTo` closures)

File: src/examples/Cluster/Metrics/Samples.Cluster.Metrics.Common/FactorialBackend.cs
Comment (Aaronontheweb) [inline src/examples/Cluster/Metrics/Samples.Cluster.Metrics.Common/FactorialBackend.cs:1]: Applying `AK1001` fixes here (`PipeTo` closures)

**Review by Aaronontheweb — COMMENTED** [review]
Detailed my changes - one last thing we need to do is to ensure that the core Akka package lists this analyzer package as a dependency. That will allow Akka.Analyzers to be consumed from end-user applications automatically, which is the entire point. This PR doesn't do that. I think I'll handle that one in a separate PR later.

**Comment by Aaronontheweb** [comment]
And as far as website documentation goes: that is planned and will happen in subsequent PRs https://github.com/akkadotnet/akka.analyzers/issues/6

**Review by Arkatufus — APPROVED** [review]
LGTM
